x-pack/plugins/alerting/server/saved_objects/index.ts

```diff
@@ -30,48 +30,61 @@ import {
   MAINTENANCE_WINDOW_SAVED_OBJECT_TYPE,
 } from '../../common';
 import { ruleModelVersions } from './rule_model_versions';
+import { RuleAttributes } from '../data/rule/types';
 
 export const RULE_SAVED_OBJECT_TYPE = 'alert';
 
-// Use caution when removing items from this array! Any field which has
-// ever existed in the rule SO must be included in this array to prevent
-// decryption failures during migration.
-export const RuleAttributesExcludedFromAAD = [
-  'scheduledTaskId',
-  'muteAll',
-  'mutedInstanceIds',
-  'updatedBy',
-  'updatedAt',
-  'executionStatus',
-  'monitoring',
-  'snoozeEndTime', // field removed in 8.2, but must be retained in case an rule created/updated in 8.2 is being migrated
-  'snoozeSchedule',
-  'isSnoozedUntil',
-  'lastRun',
-  'nextRun',
-  'revision',
-  'running',
-];
+export const RuleAttributesToEncrypt = ['apiKey'];
 
-// useful for Pick<RawAlert, RuleAttributesExcludedFromAAD> which is a
-// type which is a subset of RawAlert with just attributes excluded from AAD
+// Use caution when removing items from this array! These fields
+// are used to construct decryption AAD and must be remain in
+// this array to prevent decryption failures during migration.
+export const RuleAttributesIncludedInAAD = [
+  'enabled',
+  'name',
+  'tags',
+  'alertTypeId',
+  'consumer',
+  'legacyId',
+  'schedule',
+  'actions',
+  'params',
+  'mapped_params',
+  'createdBy',
+  'createdAt',
+  'apiKeyOwner',
+  'apiKeyCreatedByUser',
+  'throttle',
+  'notifyWhen',
+  'meta',
+];
 
-// useful for Pick<RawAlert, RuleAttributesExcludedFromAAD>
-export type RuleAttributesExcludedFromAADType =
-  | 'scheduledTaskId'
-  | 'muteAll'
-  | 'mutedInstanceIds'
-  | 'updatedBy'
-  | 'updatedAt'
-  | 'executionStatus'
-  | 'monitoring'
-  | 'snoozeEndTime'
-  | 'snoozeSchedule'
-  | 'isSnoozedUntil'
-  | 'lastRun'
-  | 'nextRun'
-  | 'revision'
-  | 'running';
+// useful type for Omit<RuleAttributes, [...RuleAttributesToEncrypt, ...RuleAttributesIncludedInAAD]>
+// which will produce a subset of RuleAttributes with just attributes that are safe to partually
+// update from AAD
+export type PartiallyUpdateableRuleAttributes = Partial<
+  Omit<
+    RuleAttributes,
+    | 'apiKey'
+    | 'enabled'
+    | 'name'
+    | 'tags'
+    | 'alertTypeId'
+    | 'consumer'
+    | 'legacyId'
+    | 'schedule'
+    | 'actions'
+    | 'params'
+    | 'mapped_params'
+    | 'createdBy'
+    | 'createdAt'
+    | 'apiKeyOwner'
+    | 'apiKeyCreatedByUser'
+    | 'throttle'
+    | 'notifyWhen'
+    | 'meta'
+  >
```

Comment:
I have attempted to entirely remove all of the alert/rule definitions of "attributes excluded from AAD". This affected the [partially update rule code](http://x-pack/plugins/alerting/server/saved_objects/partially_update_rule.ts), where I had to change from using `pick` to `omit` in order to filter the SO attributes appropriately, which lead me to replace `RawRule` with `RuleAttributes`. I've run into a cascading type-related issue due to this change which seems to boil down to `RuleSnooze`/`RuleSnoozeSchedule`.
See [rule_snooze_type.ts](http://x-pack/plugins/alerting/common/rule_snooze_type.ts):

> // Type signature of has to be repeated here to avoid issues with SavedObject compatibility
> // RuleSnooze = RuleSnoozeSchedule[] throws typescript errors across the whole lib

I think I need a more experienced set of eyes with this codebase. Is anyone available to take a look? :pray: I am happy to get on zoom and explain in better detail.